include/drivers/gpio/gpio_sim.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2020 Friedt Professional Engineering Services, Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_GPIO_GPIO_SIM_H_
+#define ZEPHYR_INCLUDE_DRIVERS_GPIO_GPIO_SIM_H_
+
+#include <zephyr/types.h>
+#include <drivers/gpio.h>
+
+int gpio_sim_input_set_masked(struct device *port, gpio_port_pins_t mask,
+			      gpio_port_value_t values);
+
+static inline int gpio_sim_input_set(struct device *port, gpio_pin_t pin,
+				     int value)
+{
+	return gpio_sim_input_set_masked(port, 1 << pin, (!!value) << pin);
+}
+
+int gpio_sim_output_get_masked(struct device *port, gpio_port_pins_t mask,
+			       gpio_port_value_t *values);
+
+static inline int gpio_sim_output_get(struct device *port, gpio_pin_t pin)
+{
+	int r;
```

Comment:
How about 'ret' which avoids a single-char var name and seems to be most common in Zephyr.